andela/coding-geeks-ah-backend
# #167484597 implemented like or dislike an article

#### What does this PR do?
It implements the approach of being able to like or dislike an article.

#### Description of Task to be done?
1. Adding routes for liking or disliking an article.
2. Adding controllers for liking or disliking an article.
3. Adding tests for liking or disliking an article.
4. Documenting the feature with Swagger.

#### Screenshots
<img width="488" alt="Screen Shot 2019-08-21 at 23 13 31" src="https://user-images.githubusercontent.com/34879554/63472876-cfe9cd00-c473-11e9-8081-a9bdb5b866f3.png">
<img width="478" alt="Screen Shot 2019-08-21 at 23 13 38" src="https://user-images.githubusercontent.com/34879554/63472877-d0826380-c473-11e9-844a-8c3dcc699f28.png">
<img width="549" alt="Screen Shot 2019-08-21 at 17 21 57" src="https://user-images.githubusercontent.com/34879554/63472792-88634100-c473-11e9-82b8-666a2d6bd66d.png">
<img width="515" alt="Screen Shot 2019-08-21 at 23 13 22" src="https://user-images.githubusercontent.com/34879554/63472865-c2ccde00-c473-11e9-93d3-e7b655e88de2.png">
<img width="1382" alt="Screen Shot 2019-08-21 at 23 11 37" src="https://user-images.githubusercontent.com/34879554/63472893-db3cf880-c473-11e9-9acf-f3647d3fb5b0.png">
<img width="1425" alt="Screen Shot 2019-08-21 at 23 11 56" src="https://user-images.githubusercontent.com/34879554/63472894-dbd58f00-c473-11e9-8203-5f4739db5465.png">
<img width="1418" alt="Screen Shot 2019-08-21 at 23 12 06" src="https://user-images.githubusercontent.com/34879554/63472895-dbd58f00-c473-11e9-880b-ee0590b1d138.png">
<img width="1388" alt="Screen Shot 2019-08-21 at 23 12 26" src="https://user-images.githubusercontent.com/34879554/63472896-dc6e2580-c473-11e9-822f-d0310f7f25bc.png">



#### How should this be manually tested?
1. Clone this branch.
2. Do **npm install**.
4. Do **npm start**
5. Create an article.
6. Like it with the following endpoint: **/api/v1/articles/{articleSlug}/like**
7. Dislike it with the following endpoint: **/api/v1/articles/{articleSlug}/dislike**
8. Retrieve the articles and check the total likes and dislikes.
